examples/asym_auth.c

```diff
@@ -0,0 +1,208 @@
+  yh_rc yrc = YHR_GENERIC_ERROR;
+
+  const char *connector_url = getenv("DEFAULT_CONNECTOR_URL");
+  if (connector_url == NULL) {
+    connector_url = DEFAULT_CONNECTOR_URL;
+  }
+
+  yrc = yh_init();
+  if (yrc != YHR_SUCCESS) {
+    fprintf(stderr, "unable to initialize yubihsm: %s\n", yh_strerror(yrc));
+    exit(EXIT_FAILURE);
+  }
+
+  yrc = yh_init_connector(connector_url, &connector);
+  assert(yrc == YHR_SUCCESS);
+
+  yrc = yh_connect(connector, 0);
+  assert(yrc == YHR_SUCCESS);
+
+  char *received_url;
+  yrc = yh_get_connector_address(connector, &received_url);
+  assert(yrc == YHR_SUCCESS);
+
+  yh_set_verbosity(connector, YH_VERB_ALL);
+
+  uint8_t d_major, d_minor, d_patch;
+  uint32_t serial;
+  uint8_t log_total, log_used;
+  yh_algorithm algorithms[YH_MAX_ALGORITHM_COUNT];
+  size_t n_algorithms = sizeof(algorithms);
+  yrc =
+    yh_util_get_device_info(connector, &d_major, &d_minor, &d_patch, &serial,
+                            &log_total, &log_used, algorithms, &n_algorithms);
+  assert(yrc == YHR_SUCCESS);
+
+  yh_algorithm key = YH_ALGO_EC_P256_YUBICO_AUTHENTICATION;
+  if (!bsearch(&key, algorithms, n_algorithms, sizeof(key),
+               compare_algorithm)) {
+    fprintf(stderr, "Skipping this test because the device does not support "
+                    "aymmetric authentication\n");
+    exit(EXIT_SUCCESS);
+  }
+
+  printf("Send a plain (unencrypted, unauthenticated) echo command\n");
+
+  uint16_t data_len = sizeof(data) - 1;
+  uint8_t response[sizeof(data)] = {0};
+  size_t response_len = sizeof(response);
+  yh_cmd response_cmd;
+  yrc = yh_send_plain_msg(connector, YHC_ECHO, data, data_len, &response_cmd,
+                          response, &response_len);
+  if (yrc != YHR_SUCCESS) {
+    fprintf(stderr, "Failed to send ECHO command): %s\n", yh_strerror(yrc));
+    exit(EXIT_FAILURE);
+  }
+
+  if (response_cmd == YHC_ERROR) {
+    fprintf(stderr, "Unable to get echo data: %s (%x)\n",
+            yh_strerror(response[0]), response[0]);
+    exit(EXIT_FAILURE);
+  }
+
+  printf("Response (%zu bytes): \"%s\"\n", response_len, response);
+
+  yh_session *session = NULL;
+  uint16_t authkey = 1;
+
+  yrc = yh_create_session_derived(connector, authkey, password,
+                                  sizeof(password) - 1, false, &session);
+  assert(yrc == YHR_SUCCESS);
+
+  yrc = yh_authenticate_session(session);
+  assert(yrc == YHR_SUCCESS);
+
+  authkey = 2;
+
+  yh_util_delete_object(session, authkey, YH_AUTHENTICATION_KEY);
+  // Ignore result here
+
+  uint8_t sk_oce[32], pk_oce[65];
+  yrc = yh_util_generate_ec_p256_key(sk_oce, sizeof(sk_oce), pk_oce,
+                                     sizeof(pk_oce));
+  assert(yrc == YHR_SUCCESS);
+
+  yh_capabilities caps = {{0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff}};
+  // The public key is imported without the uncompressed point marker (value
+  // 0x04), so skip the first byte
+  yrc = yh_util_import_authentication_key(session, &authkey, "EC Auth Key",
```

Comment:
As much as I like keeping the same interface for importing keys I think that overloading the meaning of `key_enc` in this way makes it more confusing than helpful. This probably needs more discussion but I think I'd rather have a new function.

Reply:
The api was a bit unfortunate to begin with, there aren't really two separate keys even for the symmetric case. So by simply concatenating the two inputs this works for both asym and sym keys. I felt that duplicating all that code was less attractive than doing this.

Reply:
I guess I could add a new function with just one input, and make the old one just call that

Reply:
I think that hiding the ugliness behind a more descriptive public API would be better.

Reply:
Im not sure what you mean by that, do you agree that we should add a new function with just one input ?

Reply:
I do.

Reply:
Fixed in 5c468798663571d163a4a642f57f2b6451b8c385

Reply:
what does `ext` stand for? External? Extended? When we updated the API for v2 we tried to spell things out as much as possible.